app/demo-geant-integration/demo-geant-integration.cc

```diff
@@ -0,0 +1,128 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2022 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file demo-geant-integration/demo-geant-integration.cc
+//---------------------------------------------------------------------------//
+
+#include <cstddef>
+#include <iostream>
+// #include <FTFP_BERT.hh>
+#include <CLHEP/Random/Random.h>
+#include <G4RunManager.hh>
+#include <G4Threading.hh>
+#include <G4UImanager.hh>
+
+#include "celeritas/ext/GeantVersion.hh"
+#if !CELERITAS_G4_V10
+#    include <G4RunManagerFactory.hh>
+#else
+#    include <G4MTRunManager.hh>
+#endif
+
+#include "corecel/Assert.hh"
+#include "corecel/io/Logger.hh"
+#include "corecel/sys/Environment.hh"
+#include "celeritas/ext/detail/GeantPhysicsList.hh"
+#include "accel/Logger.hh"
+
+#include "ActionInitialization.hh"
+#include "DetectorConstruction.hh"
+#include "GlobalSetup.hh"
+#include "PrimaryGeneratorAction.hh"
+
+namespace
+{
+//---------------------------------------------------------------------------//
+int GetNumThreads()
+{
+    const std::string& nt_str = celeritas::getenv("NUM_THREADS");
+    if (!nt_str.empty())
+    {
+        try
+        {
+            return std::stoi(nt_str);
+        }
+        catch (const std::logic_error& e)
+        {
+            std::cerr << "error: failed to parse NUM_THREADS='" << nt_str
+                      << "' as an integer" << std::endl;
+            return -1;
+        }
+    }
+    return G4Threading::G4GetNumberOfCores();
+}
```

Comment:
@whokion I know your `accNN` demos have a `-t` command line option to set the number of threads, but is that necessary when Geant4 (according to a "multithreading" presentation I saw) has an option to set the threads with the environment variables or the UI?

Reply:
Yes, it can be set through the UI command that the run category provides.

Reply:
Actually it's not clear to me how we could instantiate the run manager with MT vs Serial without our own environment variable... :\

Reply:
With Geant4 11.0+ (Tasking instead of MT), the serial is essentially same as  numberOfThreads=1.  I had a different motivation of having -t option (during the transition from 10.7 to 11.0), but it is indeed not needed.